app.js

```diff
@@ -1,19 +1,32 @@
-const express = require('express');
-const { getEndpoints } = require('./controllers/endpoints.controller');
-const getTopics = require('./controllers/topics.controller')
+const express = require("express");
+const { getEndpoints } = require("./controllers/endpoints.controller");
+const getTopics = require("./controllers/topics.controller");
+const getArticleById = require("./controllers/articles.controller");
 const app = express();
 
-app.get('/api', getEndpoints);
+app.get("/api", getEndpoints);
 
-app.get('/api/topics', getTopics);
+app.get("/api/topics", getTopics);
+
+app.get("/api/articles/:article_id", getArticleById);
 
 app.all("/*", (req, res, next) => {
-	res.status(404).send({message: "Invalid path"})
-})
+  res.status(404).send({ message: "Invalid path" });
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ message: "Bad request" });
+  } else next(err);
+});
+
+app.use((err, req, res, next) => {
+  res.status(404).send({ message: "Item not found" });
+});
```

Comment:
I think at this point, having a custom error handler is going to help you much more :)